CRM/Utils/Date.php

```diff
@@ -161,20 +161,18 @@ public static function format($date, $separator = '', $invalidDate = 0) {
    */
   public static function getAbbrWeekdayNames() {
     $key = 'abbrDays_' . \CRM_Core_I18n::getLocale();
-    $days = &\Civi::$statics[__CLASS__][$key];
-    if (!$days) {
-      $days = [];
-      // First day of the week
-      $firstDay = Civi::settings()->get('weekBegins');
-
-      // set LC_TIME and build the arrays from locale-provided names
-      // June 1st, 1970 was a Monday
-      CRM_Core_I18n::setLcTime();
-      for ($i = $firstDay; count($days) < 7; $i = $i > 5 ? 0 : $i + 1) {
-        $days[$i] = strftime('%a', mktime(0, 0, 0, 6, $i, 1970));
-      }
+    if (empty(\Civi::$statics[__CLASS__][$key])) {
```

Comment:
I think this implementation would break the "first day of week" setting. Previously it indexed the array differently in order to implement that. Now it always starts on Sunday.

Reply:
good point @demeritcowboy I have pushed a fix for that and I have added a unit test to lock in that change